apps/wallet/src/ui/app/components/transactions-card/Transaction.tsx

```diff
@@ -0,0 +1,216 @@
+        return isSuTransfer || txnKind === 'TransferObject'
+            ? isSender
+                ? 'To'
+                : 'From'
+            : 'Action';
+    }, [isSender, isSuTransfer, txnKind]);
+
+    return (
+        <Link
+            to={`/receipt?${new URLSearchParams({
+                txdigest: txn.certificate.transactionDigest,
+            }).toString()}`}
+            className="flex items-center w-full flex-col gap-2 py-4 no-underline"
+        >
+            <div className="flex items-start w-full justify-between gap-3">
+                <div className="w-7.5">
+                    <TxnIcon
+                        txnKindName={txnIconName}
+                        txnFailed={executionStatus === 'Failed' || !!error}
+                        isSender={isSender}
+                    />
+                </div>
+                <div className="flex flex-col w-full gap-1.5">
+                    {error ? (
+                        <div className="flex flex-col w-full gap-1.5">
+                            <Text color="gray-90" weight="semibold">
+                                Transaction Failed
+                            </Text>
+                            <div className="flex break-all text-issue-dark text-subtitle">
+                                {error}
+                            </div>
+                        </div>
+                    ) : (
+                        <>
+                            <div className="flex w-full justify-between flex-col gap-1">
+                                <div className="flex w-full justify-between ">
+                                    <div className="flex gap-1 align-middle  items-baseline">
+                                        <Text color="gray-90" weight="semibold">
+                                            {isMint
+                                                ? 'Minted'
+                                                : isSender
+                                                ? 'Sent'
+                                                : 'Received'}
+                                        </Text>
+                                        {isSuTransfer && (
+                                            <Text
+                                                color="gray-90"
+                                                weight="normal"
+                                                variant="subtitleSmall"
+                                            >
+                                                SUI
+                                            </Text>
+                                        )}
+                                    </div>
+
+                                    <CoinBalance amount={amount} />
+                                </div>
+                                <div className="flex flex-col w-full">
+                                    <div className="flex flex-col w-full gap-1.5">
+                                        <TxnTypeLabel
```

Comment:
Why not do the label computation in this component instead of hoisting this logic to the parent component?

Reply:
done